File: src/main/java/com/wikia/tibia/gateways/LootGateway.java
package com.wikia.tibia.gateways;

// TODO does this even need to extend WikiObjectGateway, as Loot does not extend WikiObject ?
public class LootGateway extends WikiObjectGateway {

    public LootGateway() {
        super(Contracts.LOOT_STATISTICS_V2);
    }
}

File: src/main/kotlin/com/wikia/tibia/gateways/LootGateway.kt
package com.wikia.tibia.gateways

import com.google.common.net.UrlEscapers
import com.wikia.tibia.enums.Contract
import com.wikia.tibia.http.Request
import io.vavr.control.Try

class LootGateway {

    private val contracts = Contract.LOOT_STATISTICS_V2
    private val request: Request = Request()

    val loot: Try<String>
        get() = getLoot(false)

    fun getLoot(expand: Boolean): Try<String> {
        return request.get("${contracts.description}?expand=$expand")
    }

    fun getLoot(pageName: String): Try<String> {
        return request.get("${contracts.description}/${UrlEscapers.urlFragmentEscaper().escape(pageName)}")
    }
}